relay-server/src/services/projects/cache/project.rs

```diff
@@ -0,0 +1,299 @@
+use std::sync::Arc;
+
+use relay_config::Config;
+use relay_dynamic_config::Feature;
+use relay_quotas::{CachedRateLimits, DataCategory, MetricNamespaceScoping, RateLimits};
+use relay_sampling::evaluation::ReservoirCounters;
+
+use crate::envelope::ItemType;
+use crate::services::outcome::{DiscardReason, Outcome};
+use crate::services::projects::cache::state::SharedProject;
+use crate::services::projects::project::ProjectState;
+use crate::utils::{CheckLimits, Enforcement, EnvelopeLimiter, ManagedEnvelope};
+
+/// A loaded project.
+pub struct Project<'a> {
+    shared: SharedProject,
+    config: &'a Config,
+}
+
+impl<'a> Project<'a> {
+    pub(crate) fn new(shared: SharedProject, config: &'a Config) -> Self {
+        Self { shared, config }
+    }
+
+    /// Returns a reference to the currently cached project state.
+    pub fn project_state(&self) -> &ProjectState {
```

Comment:
nit: redundant prefix.

```suggestion
    pub fn state(&self) -> &ProjectState {
```